src/core/core.controller.js

```diff
@@ -627,8 +627,13 @@ module.exports = function(Chart) {
 			}
 
 			// On Hover hook
-			if (hoverOptions.onHover) {
-				hoverOptions.onHover.call(me, me.active);
+			if (e.type === 'mousemove') {
+			    if (hoverOptions.onHover) {
+			        hoverOptions.onHover.call(me, me.active);
+			    }
+			    if (me.legend && me.legend.handleEvent && me.legend.options.onHover) {
```

Comment:
I see. Thanks for pointing that out Simon. I'll update my code and submit again.